swim/member.go

```diff
@@ -128,6 +184,14 @@ func shouldProcessGossip(old *Member, gossip *Member) bool {
 		return false
 	}
 
+	// Only pick the gossip if the labels have a lower checksum then the checksum
+	// of the currently known labels
+	gossipLabelsChecksum := gossip.Labels.checksum()
+	oldLabelsChecksum := old.Labels.checksum()
+	if gossipLabelsChecksum < oldLabelsChecksum {
```

Comment:
Fair point on your update. I will change it around so we pick the highest value in all checks.

Besides I will add a second if statement to always return false on a lower number to harden for future additions in the same way that all other checks are in pairs.